tests/test_signed_payload.py

```diff
@@ -3,74 +3,120 @@
-        print base64.b64encode(signature.to_bytes())
+        re = self.client.post(
+            '/v1/test_signed_payload',
+            data=json.dumps(''),
+            headers=headers
+        )
 
-    def test_signed_payload(self):
-        payload = {
-            "service_proposal": {
-                "id": 1,
-                "format": "service-proposal/v1",
-                "provider_id": "node1",
-            }
+        self.assertEqual(401, re.status_code)
+        self.assertEqual(re.json['error'], 'missing Authorization in request header')
+
+    def test_incorrect_authorization_header_value_format(self):
+        headers = {
+            "Authorization": "Signature"
         }
 
+        re = self.client.post(
+            '/v1/test_signed_payload',
+            data=json.dumps(''),
+            headers=headers
+        )
+
+        self.assertEqual(401, re.status_code)
+        self.assertEqual(re.json['error'], 'invalid Authorization header value provided, correct format: Signature <signature_base64_encoded>')
+
+    def test_incorrect_authorization_type(self):
         headers = {
-            "identity": "0x1a642f0e3c3af545e7acbd38b07251b3990914f1",
-            "Authorization": "Signature Da1mAwK5abmXQCNsCE+YjsZbR9jTyEKqdrjxxMKwNzwr2NFnM35UiVQJWcg8rgL+X2PR60LoIUMlGU9OPaSoZwE="
+            "Authorization": "incorrectType sig"
         }
 
         re = self.client.post(
-            '/v1/signed_payload',
-            data=json.dumps(payload),
+            '/v1/test_signed_payload',
+            data=json.dumps(''),
             headers=headers
         )
 
-        self.assertEqual({}, re.json)
-        self.assertEqual(200, re.status_code)
+        self.assertEqual(401, re.status_code)
+        self.assertEqual(re.json['error'], 'authentication type have to be Signature')
 
-    def test_incorrectly_signed_payload(self):
-        payload = {}
+    def test_empty_authorization_header_value(self):
+        headers = {
+            "Authorization": "Signature "
+        }
+
+        re = self.client.post(
+            '/v1/test_signed_payload',
+            data=json.dumps(''),
+            headers=headers
+        )
 
+        self.assertEqual(401, re.status_code)
+        self.assertEqual(re.json['error'], 'signature was not provided')
+
+    def test_signature_not_base64_encoded(self):
         headers = {
-            "identity": "0x0000000000000000000000000000000000000001",
-            "Authorization": "Signature Da1mAwK5abmXQCNsCE+YjsZbR9jTyEKqdrjxxMKwNzwr2NFnM35UiVQJWcg8rgL+X2PR60LoIUMlGU9OPaSoZwE="
+            "Authorization": "Signature not_base_64"
         }
 
         re = self.client.post(
-            '/v1/signed_payload',
+            '/v1/test_signed_payload',
+            data=json.dumps(''),
+            headers=headers
+        )
+
+        self.assertEqual(401, re.status_code)
+        self.assertEqual(re.json['error'], 'signature must be base64 encoded: Incorrect padding')
```

Comment:
That's quite a lot of tests for testing decorator - maybe it would be simpler to test decorator function directly?
In that case, we could have:
- Many tests calling `recover_identity`
- Two tests calling `/v1/test_signed_payload` - one success, one with failure (any of all possible)

Reply:
Refactored. Now many tests calling decode_authorization_header(header), two tests calling /v1/test_signed_payload